types/cloneable-readable/index.d.ts

```diff
@@ -1,17 +1,22 @@
-// Type definitions for cloneable-readable 1.1
+// Type definitions for cloneable-readable 2.0
 // Project: https://github.com/mcollina/cloneable-readable#readme
-// Definitions by: Nikita Volodin <https://github.com/qlonik>
+// Definitions by: Nikita Volodin <https://github.com/qlonik>, Oleg Vaskevich <https://github.com/vaskevich>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.2
 
 /// <reference types="node"/>
 
-import { Readable } from 'stream';
+import { Readable } from "stream";
+
+declare namespace cloneable {
+    type Cloneable<T extends Readable> = T & { clone(): Cloneable<T> };
+}
 
-type Cloneable<T> = T & { clone(): Cloneable<T> };
```

Comment:
I'm not sure of the reasoning behind the change to a namespace? It seems to complicate the declaration and make type name longer. Could you elaborate please?

Reply:
This is to export `Cloneable` so it can be used as a type externally. Unfortunately writing `export type Cloneable ...` doesn't work due to this error:

```
An export assignment cannot be used in a module with other exported elements.
```

and adding a namespace seems to be the workaround.